inst/rsn/RSN_02.Rmd

```diff
@@ -0,0 +1,390 @@
+       sp.layout = eth_outline, auto.key=TRUE, key.space="right",
+       par.settings = list(axis.line = list(col =  'transparent')),
+       main="Other Animals")
+
+grid.arrange(chickens, cattle, sheep, goats, otheran, ncol=2, nrow=3)
+
+```
+
+### Making a "Real-World Network" from the data
+
+```{r distmat}
+
+distMat<-round(spDists(as.matrix(lc_eth_spdf@coords), longlat=TRUE))  # dist in km
+row.names(distMat)<-lc_eth_spdf$newid;
+colnames(distMat)<-lc_eth_spdf$newid;
+
+# Process KNOWN
+
+mat1<-setNames(melt(distMat), c('row', 'col', 'values'))
+mat1$weight<-1/(mat1$values)
+
+mat1$values[mat1$values > 50] <- 0  # No links between points > 50 km apart
+mat1$values[mat1$values != 0] <- 1  # all other links are 1
+
+# Now let's randomly rewire some 0s & 1s
+
+mat1$newvalues<-sample(c(0,1), size=nrow(mat1), replace=TRUE, prob=c(.99, 0.01))
+
+# Now we remove all the matrices where we didn't have any links, and pick out only # the connected network
+
+mat2<-mat1[mat1$newvalues==1,]; mat2$values<-NULL; mat2$newvalues<-NULL;
+
+# We will also set up vertices here
+nodes<-data.frame(row=unique(mat2$row))
+nodes<-merge(nodes, 
+             data.frame(subset(lc_eth_spdf@data, select=c("newid", "chicken"))),
+             by.x="row", by.y="newid", all=FALSE)
+
+rw <- graph_from_data_frame(d=mat2, vertices=nodes, directed=T)
+
+rws<- simplify(rw, remove.multiple = T, remove.loops = F, 
+                   edge.attr.comb=c(weight="mean", type="ignore") )
+
+```
+
+### Compare this to a Watts-Strogatz model of similar extent:
+
+```{r}
+
+ws<-watts.strogatz.game(1, nrow(lc_eth_spdf@coords), 10, 0.01)
+V(ws)$chicken<-sample(lc_eth_spdf@data$chicken, nrow(lc_eth_spdf@coords))
+
+```
+
+### Compare clustering coefficients
+
+```{r rsn_clust}
+
+data.frame(Model=c("Real World", "Small World"), 
+           Value=round(c(transitivity(rws), transitivity(ws)), 3),
+           Meaning=c("Low clustering", "Higher clustering"))
+
+```
+
+### Compare path lengths between all pairs of vertices
+
+```{r rsn_dist}
+
+data.frame(Model=c("Real World", "Small World"), 
+           Value=round(c(mean_distance(rws), mean_distance(ws)), 3))
+
+```
+
+### Compare closeness centrality
+
+Closeness centrality is defined as how many steps are required to access every other vertex from a given vertex.
+
+```{r rsn_close}
+
+data.frame(Model=c("Real World", "Small World"), 
+           Value=round(c(mean(closeness(rws)), mean(closeness((ws)))), 6))
+
+```
+
+### Compare assortativity based on chicken variable
+
+```{r rsn_assort}
+
+data.frame(Model=c("Real World", "Small World"), 
+           Value=round(c(assortativity(rws, V(rws)$chicken, directed=F),
+                         assortativity(ws, V(ws)$chicken, directed=F)), 6),
+           Meaning=c("Negative tendency of node connections where chicken values are similar", "Negative tendency of node connections where chicken values are similar"))
+
+```
+
+### Communities: Small World Network
+
```

Comment:
This is helpful.  I think we'll need a deeper dive to explain it.  We'll want to look into network _modularity_ and also check how well these clusters match up with community sizes from the survey.